cpp/src/io/json/reader_impl.cu

```diff
@@ -600,9 +601,22 @@ table_with_metadata reader::impl::convert_data_to_table(rmm::cuda_stream_view st
   stream.synchronize();
 
   // postprocess columns
-  auto target = make_strings_column(
-    std::vector<char>{'\\', '"', '\\', '\\', '\\', 't', '\\', 'r', '\\', 'b'}, {0, 2, 4, 6, 8, 10});
-  auto repl = make_strings_column({'"', '\\', '\t', '\r', '\b'}, {0, 1, 2, 3, 4, 5});
+  auto target_chars   = std::vector<char>{'\\', '"', '\\', '\\', '\\', 't', '\\', 'r', '\\', 'b'};
+  auto target_offsets = std::vector<size_type>{0, 2, 4, 6, 8, 10};
+
+  auto repl_chars   = std::vector<char>{'"', '\\', '\t', '\r', '\b'};
+  auto repl_offsets = std::vector<size_type>{0, 1, 2, 3, 4, 5};
+
+  auto target = make_strings_column(cudf::detail::make_device_uvector_async(target_chars, stream),
+                                    cudf::detail::make_device_uvector_async(target_offsets, stream),
+                                    {},
+                                    0,
+                                    stream);
+  auto repl   = make_strings_column(cudf::detail::make_device_uvector_async(repl_chars, stream),
+                                  cudf::detail::make_device_uvector_async(repl_offsets, stream),
```

Comment:
Is there a sync somewhere later to make sure this copy is done?

Reply:
Ignoring the later D2H copy without a stream, I don't think this requires a sync, since the function we are inside takes a stream, and is thus assumed to follow stream ordering semantics on that stream (and does so by passing the stream to these uvector and make_strings_column functions.

Reply:
But the `target_chars/target_offsets` and `repl_chars/repl_offsets` are temporary local vectors that will get destroyed when the function exits and the copy may not have completed yet. Or am I missing something?

Reply:
Changing the structure is out of scope of this PR, so I added a sync for now with a comment explaining why.